docs: clarify prompt caching semantics in Python (#4499)
langchain-ai/docs: src/oss/deepagents/harness.mdx

### Prompt caching

> For Anthropic models, `create_deep_agent` automatically applies prompt caching to the stable sections of the system prompt. The base agent instructions and skill content are static or mostly stable across turns, so caching them avoids reprocessing the same tokens across calls, reducing both latency and cost on long-running agents. Dynamic sections such as memory may change between turns; the harness and middleware order content to reduce cache invalidation where possible.

Author: ```suggestion
For Anthropic models, `create_deep_agent` automatically applies prompt caching to static sections of the system prompt—the base agent instructions, memory, and skill content that repeat on every turn. This avoids reprocessing the same tokens across calls, reducing both latency and cost on long-running agents. Dynamic sections such as memory may change between turns; the harness and middleware order content to reduce cache invalidation where possible.
```

Author: This suggestion was already reflected in the branch before the latest commit, so no additional change was needed there.
